pkgs/dart_mcp/test/client_and_server_test.dart

```diff
@@ -205,6 +205,89 @@ void main() {
     await environment.serverConnection.shutdown();
     expect(environment.client.connections, isEmpty);
   });
+
+  group('version negotiation', () {
+    test('server can downgrade the version', () async {
+      final environment = TestEnvironment(
+        TestMCPClient(),
+        (c) => TestOldMcpServer(channel: c),
+      );
+
+      final initializeResult = await environment.initializeServer();
+      expect(
+        initializeResult.protocolVersion,
+        ProtocolVersion.oldestSupportedVersion,
+      );
+    });
+
+    test('server can accept a lower version', () async {
+      final environment = TestEnvironment(
+        TestMCPClient(),
+        (c) => TestMCPServer(channel: c),
+      );
+      final initializeResult = await environment.initializeServer(
+        protocolVersion: ProtocolVersion.oldestSupportedVersion,
+      );
+      expect(
+        initializeResult.protocolVersion,
+        ProtocolVersion.oldestSupportedVersion,
+      );
+    });
+
+    test(
+      'client will shut down the server if version negotiation fails',
+      () async {
+        final environment = TestEnvironment(
+          TestMCPClient(),
+          (c) => TestUnrecognizedVersionMcpServer(channel: c),
+        );
+        await environment.initializeServer();
+        expect(environment.client.connections, isEmpty);
+        expect(environment.serverConnection.isActive, false);
+      },
+    );
+
+    test('protocol versions can be compared', () {
```

Comment:
can we break this out into a separate test library api_test.dart to match where the enum is defined?

Reply:
done